gpkit/model.py

```diff
@@ -1,250 +1,424 @@
 # -*- coding: utf-8 -*-
-"""Module for creating models.
+"""Module for creating Model instances.
 
-    Currently these are only used for GP instances, but they may be further
-    abstractable.
+    Example
+    -------
+    >>> gp = gpkit.Model(cost, constraints, substitutions)
 
 """
 
 import numpy as np
 
+from time import time
+from pprint import pformat
+from collections import defaultdict
 from functools import reduce as functools_reduce
-from operator import add
+from operator import mul, add
+from copy import deepcopy
 from collections import Iterable
 
-from .small_classes import Strings
-from .small_classes import PosyTuple
-from .small_classes import CootMatrix
-from .nomials import Constraint, Monomial
+from .nomials import Constraint, MonoEQConstraint
+from .nomials import Posynomial, Monomial, Signomial
 from .varkey import VarKey
+from .substitution import substitution, getsubs
 
-from .substitution import substitution
+from .small_classes import Strings
+from .small_scripts import flatten
 from .small_scripts import locate_vars
-from .small_scripts import is_sweepvar
 from .small_scripts import mag
+from .small_scripts import is_sweepvar
+from .small_scripts import sort_and_simplify
+
+from .solution_array import SolutionArray
+from .signomial_program import SignomialProgram
+from .geometric_program import GeometricProgram
+
+from .variables import Variable
+
+try:
+    from IPython.parallel import Client
+    CLIENT = Client(timeout=0.01)
+    assert len(CLIENT) > 0
+    POOL = CLIENT[:]
+    POOL.use_dill()
+    print("Using parallel execution of sweeps on %s clients" % len(CLIENT))
+except:
+    POOL = None
 
 
 class Model(object):
-    "Abstract class with substituion, loading / saving, and p_idx/A generation"
```

Comment:
Absolutely need to add a good docstring for Model, since it's the public-facing API now.